core/bentley/src/StatusCategory.ts

```diff
@@ -0,0 +1,355 @@
+    case IModelHubStatus.iModelIsLocked: return new Locked();
+    case IModelHubStatus.CodesExist: return new Conflict();
+    case IModelHubStatus.LocksExist: return new Conflict();
+    case IModelHubStatus.iModelAlreadyExists: return new Conflict();
+    case IModelHubStatus.iModelDoesNotExist: return new NotFound();
+    case IModelHubStatus.FileDoesNotExist: return new NotFound();
+    case IModelHubStatus.FileAlreadyExists: return new Conflict();
+    case IModelHubStatus.LockDoesNotExist: return new NotFound();
+    case IModelHubStatus.LockOwnedByAnotherBriefcase: return new Conflict();
+    case IModelHubStatus.CodeStateInvalid: return new StateViolation();
+    case IModelHubStatus.CodeReservedByAnotherBriefcase: return new Conflict();
+    case IModelHubStatus.CodeDoesNotExist: return new NotFound();
+    case IModelHubStatus.EventTypeDoesNotExist: return new NotFound();
+    case IModelHubStatus.EventSubscriptionDoesNotExist: return new NotFound();
+    case IModelHubStatus.EventSubscriptionAlreadyExists: return new StateViolation();
+    case IModelHubStatus.ITwinIdIsNotSpecified: return new ValidationError();
+    case IModelHubStatus.FailedToGetITwinPermissions: return new OperationFailed();
+    case IModelHubStatus.FailedToGetITwinMembers: return new OperationFailed();
+    case IModelHubStatus.ChangeSetAlreadyHasVersion: return new Conflict();
+    case IModelHubStatus.VersionAlreadyExists: return new Conflict();
+    case IModelHubStatus.JobSchedulingFailed: return new InternalError();
+    case IModelHubStatus.ConflictsAggregate: return new Conflict();
+    case IModelHubStatus.FailedToGetITwinById: return new OperationFailed();
+    case IModelHubStatus.DatabaseOperationFailed: return new OperationFailed();
+    case IModelHubStatus.SeedFileInitializationFailed: return new OperationFailed();
+    case IModelHubStatus.FailedToGetAssetPermissions: return new OperationFailed();
+    case IModelHubStatus.FailedToGetAssetMembers: return new OperationFailed();
+    case IModelHubStatus.ITwinDoesNotExist: return new NotFound();
+    case IModelHubStatus.LockChunkDoesNotExist: return new NotFound();
+    case IModelHubStatus.CheckpointAlreadyExists: return new Conflict();
+    case IModelHubStatus.CheckpointDoesNotExist: return new NotFound();
+    case IModelHubStatus.UndefinedArgumentError: return new ValidationError();
+    case IModelHubStatus.InvalidArgumentError: return new ValidationError();
+    case IModelHubStatus.MissingDownloadUrlError: return new ValidationError();
+    case IModelHubStatus.NotSupportedInBrowser: return new NotSupported();
+    case IModelHubStatus.FileHandlerNotSet: return new NotImplemented();
+    case IModelHubStatus.FileNotFound: return new NotFound();
+    case IModelHubStatus.InitializationTimeout: return new Timeout();
+
+    case AuthStatus.Success: return new Success();
+    case AuthStatus.Error: return new UnknownError();
```

Comment:
Is `AuthStatus` about authentication or authorization?  If so, I would think that AuthStatus would be a 401 or a 403, respectively.

Reply:
The only uses I found were here and here:

https://github.com/iTwin/itwinjs-core/blob/cc2abab4f035283a66a0e21a9e6590932769baea/core/mobile/src/backend/MobileAuthorizationBackend.ts#L65

https://github.com/iTwin/itwinjs-core/blob/cc2abab4f035283a66a0e21a9e6590932769baea/core/mobile/src/backend/MobileAuthorizationBackend.ts#L70

AuthStatus.Error seems to be used to indicate issues encountered when configuring the auth system, not user auth itself. Maybe my ValidationError is a better fit.

Success is not used right now, as far as I can tell.

Reply:
The second usage shows that the expected URL is the issuer URL.  On the client side, I would expect to see a 401 if I messed up the auth provider configuration.  Maybe @laurynasr could chime in with his thoughts here.

The first usage is because the configuration is completely missing.  I suppose ValidationError would work well there.

Reply:
This is server side, thus `UnknownError` (i.e. 500 InternalServerError) seems a good fit to me for reporting misconfiguration.